src/webview/src/components/summary/table.js

```diff
@@ -0,0 +1,247 @@
+      prev[resKey].push(row)
+      return prev
+    }, {})
+  }
+
+  const defaultColumn = React.useMemo(
+    () => ({
+      // Let's set up our default Filter UI
+      Filter: DefaultColumnFilter
+    }),
+    []
+  );
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    preGlobalFilteredRows,
+    setGlobalFilter,
+    toggleAllRowsExpanded,
+    expandAllSubComponents,
+    state: { globalFilter },
+  } = useTable({
+    columns,
+    data,
+    defaultColumn,
+    initialState: {
+      groupBy: ['effect', 'category']
+    },
+    filterTypes,
+    globalFilter: 'customFilter',
+    groupByFn: customGroupBy,
+    autoResetPage: false,
+    autoResetExpanded: false,
+  },
+    useGlobalFilter,
+    useFilters,
+    useGroupBy,
+    useExpanded
+  )
+
+  /* https://github.com/tannerlinsley/react-table/issues/2404#issuecomment-644917151 */
+  React.useEffect(() => {
+    toggleAllRowsExpanded(expandAllSubComponents);
+  }, [expandAllSubComponents, toggleAllRowsExpanded]);
+
+  return (
+    <TableContainer>
+      <TableToolbar
+        preGlobalFilteredRows={preGlobalFilteredRows}
+        setGlobalFilter={setGlobalFilter}
+        globalFilter={globalFilter}
+        onClose={onClose}
+      />
+      <MaUTable {...getTableProps()}>
+        <TableHead>
+          {headerGroups.map(headerGroup => (
+            <TableRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => (
+                <TableCell className={classes.tableCell} {...column.getHeaderProps()}>
+                  <span style={{ whiteSpace: 'nowrap' }}>
+                    {column.render('Header')}
+                    {column.canGroupBy ? (
+                      // If the column can be grouped, let's add a toggle
+                      <span {...column.getGroupByToggleProps()}>
+                        {column.isGrouped ? <ViewHeadlineIcon style={{ verticalAlign: 'middle' }} /> : <ViewQuiltIcon style={{ verticalAlign: 'middle' }} />}
+                      </span>
+                    ) : null}
+                  </span>
+                  <div>{column.canFilter ? column.render('Filter') : null}</div>
+                </TableCell>
+              ))}
+            </TableRow>
+          ))}
+        </TableHead>
+        <TableBody {...getTableBodyProps()}>
+          {rows.map((row, i) => {
+            prepareRow(row)
+            return (
+              <React.Fragment>
+                <TableRow hover={true} {...row.getRowProps()}>
+                  {row.cells.map(cell => {
+                    return (
+                      <TableCell className={classes.tableCell} style={{ whiteSpace: 'pre-line' }} {...cell.getCellProps()}>
+                        {cell.isGrouped ? (
+                          <div>
+                            <span {...row.getToggleRowExpandedProps()}>
+                              {row.isExpanded ? '➖' : '➕'}
```

Comment:
the '-' and '+' are some special UTF-8 chars? (not the usual -/+ or?)

Reply:
Yes it's [U+2795](https://unicode.org/emoji/charts/full-emoji-list.html#2795) and [U+2796](https://unicode.org/emoji/charts/full-emoji-list.html#2796) which is from the Emoji List and used as very simple buttons on the table.